### ISSUE TITLE
Need an automated spam labeler, enabled by default.

### ISSUE THREAD

**Issue by enn-nafnlaus** [bluesky-social/atproto#3134]
**Is your feature request related to a problem? Please describe.**

This is getting out of hand.  Spammers  just show up en masse in response to keywords or links, like GoFundMe URLs.

https://bsky.app/profile/nafnlaus.bsky.social/post/3lc3hesbacs2h

Another common type is the "Gaza Bot" issue, where a bot pretending to be a Gazan needing funds jumps into random threads on random topics and/or sends random DMs.

I've already personally seen friends driven off the app by this spam. And it's quite predictable.

**Describe the solution you'd like**

An effective spam labeler (LLM-based would be fine, I can help with that if you need it; LLMs are quite capable of e.g. detecting whether someone is promoting an unrelated "fundraiser" in response to an existing one, or whether a fundraiser is being posted to an entirely unrelated topic, or whether a DM involves a fundraiser), of course factoring in factors like whether you follow the person, have been recently communicating with them and their posts weren't flagged then, etc.

It should be enabled by default for all users.

**Describe alternatives you've considered**

Human curated labelers will never be able to outpace bot registration, and they can see when they've been labeled and thus when they need to start a new account.

If it's not enabled by default, many new users won't know it exists, will see a bunch of spam that others don't see, will get frustrated, and leave.

**Additional context**

![image](https://github.com/user-attachments/assets/5457c929-7985-4cdb-9ff5-6abd6f179443)

![image](https://github.com/user-attachments/assets/7013a17e-1dda-4440-9712-1e8a3226bab1)

![image](https://github.com/user-attachments/assets/1f05dffd-e0dc-4ee4-8b4c-bbae6b419b27)

**Comment by enn-nafnlaus** [comment]
Lol, the spambots are now replying to spambots.

https://bsky.app/profile/ericktoth.bsky.social/post/3lc3nyskqgk2c

![image](https://github.com/user-attachments/assets/7ea21942-3d82-4184-9823-e1754a1e76b6)

**Comment by Zero3K** [comment]
![image](https://github.com/user-attachments/assets/18e616a7-0bc0-4907-b98d-a8ae9e6f1053)

**Comment by enn-nafnlaus** [comment]
> ![image](https://private-user-images.githubusercontent.com/6194332/391120766-18e616a7-0bc0-4907-b98d-a8ae9e6f1053.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5MDA3ODEsIm5iZiI6MTczMjkwMDQ4MSwicGF0aCI6Ii82MTk0MzMyLzM5MTEyMDc2Ni0xOGU2MTZhNy0wYmMwLTQ5MDctYjk4ZC1hOGFlOWU2ZjEwNTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjlUMTcxNDQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDI4MGIzNGMzYTFmMmU2MGE5MzAwNjZmNDhjYWUwMWFkZDU3OTk5MDJjOGZmMjRkY2ZkNTJiNjE0ZjA3ZGQ5MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.UuSbXO9R3QDd7rlRORbTWb9mUvde_sCQb9jK-nYxRc4)

Those are my settings as well.

If we're going to wait for humans to report them, and then wait for humans to review them, and only then will they disappear, then the bots - which operate instantaneously - will always win.

**Comment by Zero3K** [comment]
Makes sense then. Someone should create a labeler that can automatically detect those and other kinds of spam posts.

**Comment by enn-nafnlaus** [comment]
I'll offer to assist on the ML side if anyone else wants to do the ATProto side (or we could just do string matching, but that's sort of playing a game of whack-a-mole with spammers, since they merely need to change up the text on their end, and could even have a LLM make custom text for each message - but still, that might be a good starting point?).  

For a ML version, there's three options really - 

1) Small premade spam models (there's lots on Huggingface, but they're usually designed for email or SMS, e.g. they usually just look at the current message and thus may not catch all the nuance we may want to take into account of the context to which replies are being made)
2) Large generalist models (but they involve significant compute per message processed)
3) Custom trained small models (I'd either need a dataset of real spam and non-spam, or would need to make a synthetic dataset of simulated spam/non-spam, and can then train the model on that).

**Comment by Zero3K** [comment]
Maybe you could link to this issue in a post on your Bluesky account in order to reach more people thereby getting it made quicker.

**Comment by enn-nafnlaus** [comment]
I think you overestimate my account's reach  ;)

I did write a post mentioning the problem and the ticket, though.
